### DISCUSSION TITLE
Offline Viewing

### DISCUSSION THREAD

**Discussion by me-dex** [jellyfin/jellyfin-web#6368] [Ideas]
Re feature request: https://features.jellyfin.org/posts/3012/jellyfin-web-offline-viewing

Enable offline browsing of libraries/viewing of content downloaded from a Jellyfin server. I've been contemplating this feature for a while now and I think the web standards have finally caught up to a point where it's feasible. A combination of persistent storage APIs like [StorageManager](https://developer.mozilla.org/en-US/docs/Web/API/StorageManager) and [IndexedDB](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API) allow browser apps to finally store substantial amounts of data on the orders of GBs/TBs. It also seems to me that there's no technical barrier downloading transcoded media over the raw media for smaller file sizes on device. A queuing system would likely need to be implemented client-side though to not overload the server with download requests.

https://developer.mozilla.org/en-US/docs/Web/API/Storage_API/Storage_quotas_and_eviction_criteria#does_browser-stored_data_persist

**Comment by thornbill** [comment]
Feature requests belong at https://features.jellyfin.org

**Reply by me-dex** [reply]
I do plan to implement this feature @thornbill - I'm looking for technical discussion on implementation details as I've already started looking into this, but I've also spun up a feature request to track it over there as well

https://features.jellyfin.org/posts/3012/jellyfin-web-offline-viewing

> Contributors, if you're looking for things to implement, please create a Discussion on one of the various (most relevant) repositories regarding the feature and we can adjust its status. We strongly encourage this to avoid delays and issues during the PR process, to allow the team to discuss the feature proposal in depth before implementing.

**Reply by sjorge** [reply]
Technical discussion are usually held here https://github.com/jellyfin/jellyfin-meta/discussions

**Reply by me-dex** [reply]
@sjorge But both the [features site](https://features.jellyfin.org/) and the [readme in that repo](https://github.com/jellyfin/jellyfin-meta/blob/master/README.md) state that discussions for the individual components are to be held in their respective repos; this is a change specific to the operation of the web client

> Discussions meant for individual projects are hosted inside their respective repositories.

Have the project's rules around technical discussions changed?

**Reply by mesut3000** [reply]
Hi;

I am using jellyfin server on a windows 11 based computer. I have a movie archive of several disks. I can't insert them all at once, I add them one by one to the library, but when I remove them, they are deleted from the library. I realized that this is how the system works online, but I don't want my data to be deleted, I want to be able to see it when there is no disk, and I want to be able to play it when I insert the disk.

**Reply by thornbill** [reply]
@mesut3000 that is unrelated to this discussion
